src/Lithnet.GoogleApps/GmailServiceRequestFactory.cs

```diff
@@ -78,22 +89,52 @@ public IEnumerable<string> GetDelegates(string id)
 
         public IEnumerable<string> GetSendAsAddresses(string id)
         {
-            id.ThrowIfNotEmailAddress();
 
-            GmailService service = this.GetService(id);
-            UsersResource.SettingsResource.SendAsResource.ListRequest request = service.Users.Settings.SendAs.List(id);
-            ListSendAsResponse result = request.ExecuteWithRetry(RetryEvents.BackoffOAuthNotFound, this.RetryCount);
-            return result.SendAs?.Select(t => t.SendAsEmail) ?? new List<string>();
+            try
+            {
+                id.ThrowIfNotEmailAddress();
+
+                GmailService service = this.GetService(id, DefaultTimeout);
+                UsersResource.SettingsResource.SendAsResource.ListRequest request = service.Users.Settings.SendAs.List(id);
+                ListSendAsResponse result = request.ExecuteWithRetry(RetryEvents.Backoff | RetryEvents.Timeout, this.RetryCount);
+                return result.SendAs?.Select(t => t.SendAsEmail) ?? new List<string>();
+            }
+            catch(GoogleApiException ex)
+            {
+                if (ex.HttpStatusCode == HttpStatusCode.BadRequest && ex.Message != null && ex.Message.IndexOf("Mail service not enabled", StringComparison.OrdinalIgnoreCase) >= 0)
+                {
+                    return new List<string>();
+                }
+                else
+                {
+                    throw;
+                }
+            }
         }
 
         public IEnumerable<SendAs> GetSendAs(string id)
         {
             id.ThrowIfNotEmailAddress();
 
-            GmailService service = this.GetService(id);
-            UsersResource.SettingsResource.SendAsResource.ListRequest request = service.Users.Settings.SendAs.List(id);
-            ListSendAsResponse result = request.ExecuteWithRetry(RetryEvents.BackoffOAuthNotFound, this.RetryCount);
-            return result.SendAs ?? new List<SendAs>();
+            try
+            {
+                GmailService service = this.GetService(id, DefaultTimeout);
+                UsersResource.SettingsResource.SendAsResource.ListRequest request = service.Users.Settings.SendAs.List(id);
+                ListSendAsResponse result = request.ExecuteWithRetry(RetryEvents.Backoff | RetryEvents.Timeout, this.RetryCount);
```

Comment:
Any reason to remove the 'NotFound'/Oauth policies?

Reply:
Mistake on my side. Adding the deleted RetryEvents.